sdk/nbgl_sync.c

```diff
@@ -0,0 +1,94 @@
+#include "nbgl_use_case.h"
+#include "nbgl_sync.h"
+#include "nbgl_wrapper.h"
+#include "io.h"
+#include "ledger_assert.h"
+#include "os.h"
+
+static sync_nbgl_ret_t ret;
+static bool ended;
+
+
+static void choice_callback(bool confirm) {
+    if (confirm) {
+        ret = NBGL_SYNC_RET_SUCCESS;
+    } else {
+        ret = NBGL_SYNC_RET_REJECTED;
+    }
+
+    ended = true;
+}
+
+static void quit_callback(void) {
+    ret = NBGL_SYNC_RET_SUCCESS;
+    ended = true;
+}
+
+static void sync_nbgl_init(void) {
+   ended = false;
+   ret = NBGL_SYNC_RET_ERROR;
+}
+
+static sync_nbgl_ret_t sync_nbgl_wait(void) {
+    int apdu_state = G_io_app.apdu_state;
+    while (!ended) {
+
+        // Check for a apdu_state change.
+        // This would means an APDU was received.
+        // In such case immediately stop the loop.
+        // We could expose another version that would ignore such APDU event.
+        // Anyway, it will be handled only at next app call of io_recv_command().
+        if (apdu_state == APDU_IDLE && G_io_app.apdu_state != APDU_IDLE) {
+            return NBGL_SYNC_RET_RX_APDU;
+        }
```

Comment:
This was a proposal to interrupt sync nbgl calls like `nbgl_useCaseStatus`, or `nbgl_useCaseHome` during which there is no transaction being processed.
But this might indeed disappear as we might want to respond to the APDU only after the timeout of `nbgl_useCaseStatus` and I'm not sure we will make a sync version of `nbgl_useCaseHome`